lib/cylc/rundb.py

```diff
@@ -676,12 +683,17 @@ def select_task_pool_for_restart(self, callback, id_key=None):
             r"LEFT OUTER JOIN " +
             r"    %(task_timeout_timers)s " +
             r"ON  %(task_pool)s.cycle == %(task_timeout_timers)s.cycle AND " +
-            r"    %(task_pool)s.name == %(task_timeout_timers)s.name")
+            r"    %(task_pool)s.name == %(task_timeout_timers)s.name " +
+            r"LEFT OUTER JOIN " +
+            r"    %(task_outputs)s " +
+            r"ON  %(task_pool)s.cycle == %(task_outputs)s.cycle AND " +
+            r"    %(task_pool)s.name == %(task_outputs)s.name")
```

Comment:
I must say that I have always found this sort of thing surprising (or is it disappointing?) with Python. Surely, Python should optimise this away at compile time...